### PR TITLE
raftstore: add metrics for observing apply for 3.0.5.hotfix

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: Liqi Geng <gengliqiii@gmail.com>

### What problem does this PR solve?

Cherry-pick some code of https://github.com/tikv/tikv/pull/7702

### What is changed and how it works?

What's Changed:

Make yield and reschedule counter configurable
Add more metrics

### Related changes

No.

### Check List 

Tests 

- No code

Side effects

No.

### Release note 
* No release note

### CODE REVIEW

File: src/raftstore/store/fsm/apply.rs [hunk 1]
```diff
@@ -2623,8 +2642,15 @@ impl ApplyFsm {
             }
         }
         if let Some(timer) = channel_timer {
-            let elapsed = duration_to_sec(timer.elapsed());
-            APPLY_TASK_WAIT_TIME_HISTOGRAM.observe(elapsed);
+            let elapsed = timer.elapsed();
```
Comment (innerr) [inline src/raftstore/store/fsm/apply.rs:2645]: Why not `duration_to_sec` here?

File: src/raftstore/store/fsm/apply.rs [hunk 2]
```diff
@@ -2652,6 +2678,19 @@ impl Fsm for ApplyFsm {
     {
         self.mailbox.take()
     }
+
+    #[inline]
+    fn after_scheduled(&mut self) {
```
Comment (innerr) [inline src/raftstore/store/fsm/apply.rs:2683]: I can't find where call this func

File: src/raftstore/store/fsm/apply.rs (hunk 1 above)
Comment (gengliqi) [reply src/raftstore/store/fsm/apply.rs:2645]: Because it is used to print log

**Pull request merged**